types/index.d.ts

```diff
@@ -0,0 +1,565 @@
+/** set when v86 is built with debug mode enabled */
+declare const DEBUG: boolean;
+
+declare type V86Image = 
+  | { url: string } 
+  | { buffer: ArrayBuffer }
+  | { url: string, async: false, size?: number }
+  | { url: string, async: true, size: number }
+
+/**
+ * A 9p filesystem is supported by the emulator, using a virtio transport. Using it, files can be exchanged with the guest OS
+ * If `basefs` and `baseurl` are omitted, an empty 9p filesystem is created.
+ */
+declare interface Filesystem9pOptions {
+  /**
+   * json file created using [fs2json](https://github.com/copy/v86/blob/master/tools/fs2json.py). 
+   */
+  baseurl?: string,
+  
+  /**
+   * The base url is the prefix of a url from which the files are available. 
+   * For instance, if the 9p filesystem has a file `/bin/sh`, that file must be accessible from http://localhost/9p/base/bin/sh
+   */
+  basefs?: string
+}
+
+declare enum LogLevel {
+  LOG_ALL =    -1,
+  LOG_NONE =   0,
+  LOG_OTHER =  0x000001,
+  LOG_CPU =    0x000002,
+  LOG_FPU =    0x000004,
+  LOG_MEM =    0x000008,
+  LOG_DMA =    0x000010,
+  LOG_IO =     0x000020,
+  LOG_PS2 =    0x000040,
+  LOG_PIC =    0x000080,
+  LOG_VGA =    0x000100,
+  LOG_PIT =    0x000200,
+  LOG_MOUSE =  0x000400,
+  LOG_PCI =    0x000800,
+  LOG_BIOS =   0x001000,
+  LOG_FLOPPY = 0x002000,
+  LOG_SERIAL = 0x004000,
+  LOG_DISK =   0x008000,
+  LOG_RTC =    0x010000,
+  LOG_HPET =   0x020000,
+  LOG_ACPI =   0x040000,
+  LOG_APIC =   0x080000,
+  LOG_NET =    0x100000,
+  LOG_VIRTIO = 0x200000,
+  LOG_9P =     0x400000,
+  LOG_SB16 =   0x800000
+} 
+
+declare enum BootOrder {
+  CD_FLOPPY_HARDDISK = 0x213,
+  CD_HARDDISK_FLOPPY = 0x123,
+  FLOPPY_CD_HARDDISK = 0x231,
+  FLOPPY_HARDDISK_CD = 0x321,
+  HARDDISK_CD_FLOPPY = 0x132
+}
+
+declare class BusConnector {
+  constructor()
+  
+  /**
+   * @param {string} name
+   * @param {function(?)} fn
+   * @param {Object} this_value
+   */
+  register(name: string, fn: Function, this_value: Object);
+
+  /**
+   * Unregister one message with the given name and callback
+   *
+   * @param {string} name
+   * @param {function()} fn
+   */
+  unregister(name: string, fn: Function)
+
+  /**
+   * Send ("emit") a message
+   *
+   * @param {string} name
+   * @param {*=} value
+   * @param {*=} unused_transfer
+   */
+  send(name: string, value: any, unused_transfer?: any)
+
+  /**
+   * Send a message, guaranteeing that it is received asynchronously
+   *
+   * @param {string} name
+   * @param {Object=} value
+   */
+  send_async(name: string, value: any)
```

Comment:
This is not used, I think it should be removed (feel free to also remove it from the source).